src/core/src/package_managers/YumPackageManager.py

```diff
@@ -72,6 +79,14 @@ def invoke_package_manager(self, command):
         """Get missing updates using the command input"""
         self.composite_logger.log_debug('\nInvoking package manager using: ' + command)
         code, out = self.env_layer.run_command_output(command, False, False)
+
+        # check for known issues in the output and fix. Run the command again to get results after fix
+        if "Error" in out or "Errno" in out:
+            issue_mitigated = self.mitigate_issue(out)
+            if issue_mitigated:
+                self.composite_logger.log_debug('\nPost mitigation, invoking package manager again using: ' + command)
+                code, out = self.env_layer.run_command_output(command, False, False)
+
```

Comment:
send code and out to a function to detect issues (and return) - it may cover more cases in the future.
[normalize all functions around this into the Package Manager class for consistency in naming and usage across package managers]
- log if issues were detected
- check if the issue falls into known issues (log)
       - try to mitigate (log outcome)
- repeat check until no known issues remain (current implementation assume single issue)

Reply:
Resolved